67_RayQueryGeometry/app_resources/render.comp.hlsl

```diff
@@ -98,20 +110,49 @@ void main(uint32_t3 threadID : SV_DispatchThreadID)
             }
         }
 
-        uint32_t v0 = vk::RawBufferLoad<uint32_t>(pc.vertexBufferAddress + i0 * vertexStride + byteOffset);
-        uint32_t v1 = vk::RawBufferLoad<uint32_t>(pc.vertexBufferAddress + i1 * vertexStride + byteOffset);
-        uint32_t v2 = vk::RawBufferLoad<uint32_t>(pc.vertexBufferAddress + i2 * vertexStride + byteOffset);
+        uint32_t v0 = vk::RawBufferLoad<uint32_t>(pc.geom[instID].vertexBufferAddress + i0 * vertexStride + byteOffset);
+        uint32_t v1 = vk::RawBufferLoad<uint32_t>(pc.geom[instID].vertexBufferAddress + i1 * vertexStride + byteOffset);
+        uint32_t v2 = vk::RawBufferLoad<uint32_t>(pc.geom[instID].vertexBufferAddress + i2 * vertexStride + byteOffset);
 
-        float3 n0 = getNormalsFromMask(v0) * 0.5 + 0.5;
-        float3 n1 = getNormalsFromMask(v1) * 0.5 + 0.5;
-        float3 n2 = getNormalsFromMask(v2) * 0.5 + 0.5;
+        float3 n0, n1, n2;
+        switch (instID)
+        {
+            case OT_CUBE:
+            {
+                // this still doesn't feel right
+                // int8_t3 --> SSCALED (converts to float) --> unpackUnorm (read as uint --> float / 255)
+                n0 = (nbl::hlsl::spirv::unpackUnorm4x8(v0) * 255.0).yzw;
+                n1 = (nbl::hlsl::spirv::unpackUnorm4x8(v1) * 255.0).yzw;
+                n2 = (nbl::hlsl::spirv::unpackUnorm4x8(v2) * 255.0).yzw;
```

Comment:
you might see that the *255.0 makes no difference to the normal value because of normalization

Reply:
still a todo